mono/metadata/icall.c

```diff
@@ -707,84 +729,61 @@ ves_icall_System_Array_CreateInstanceImpl64 (MonoReflectionType *type, MonoArray
 }
 
 ICALL_EXPORT gint32 
-ves_icall_System_Array_GetRank (MonoObject *arr)
+ves_icall_System_Array_GetRank (MonoObjectHandle arr, MonoError *error)
 {
-	return m_class_get_rank (mono_object_class (arr));
+	gint32 const result = m_class_get_rank (mono_handle_class (arr));
+	icallarray_print ("%s arr:%p res:%d\n", __func__, arr, result);
+	return result;
 }
 
+ICALL_EXPORT gint64
+ves_icall_System_Array_GetLongLength (MonoArrayHandle arr, gint32 dimension, MonoError *error);
+
 ICALL_EXPORT gint32
-ves_icall_System_Array_GetLength (MonoArray *arr, gint32 dimension)
+ves_icall_System_Array_GetLength (MonoArrayHandle arr, gint32 dimension, MonoError *error)
 {
-	gint32 rank = m_class_get_rank (mono_object_class (&arr->obj));
-	uintptr_t length;
-	ERROR_DECL (error);
-
-	if ((dimension < 0) || (dimension >= rank)) {
-		mono_error_set_index_out_of_range (error);
-		mono_error_set_pending_exception (error);
-		return 0;
-	}
-	
-	if (arr->bounds == NULL)
-		length = arr->max_length;
-	else
-		length = arr->bounds [dimension].length;
-
-#ifdef MONO_BIG_ARRAYS
+	uintptr_t const length = ves_icall_System_Array_GetLongLength (arr, dimension, error);
 	if (length > G_MAXINT32) {
-		ERROR_DECL (error);
 		mono_error_set_overflow (error);
-		mono_error_set_pending_exception (error);
 		return 0;
 	}
-#endif
 	return length;
 }
 
 ICALL_EXPORT gint64
-ves_icall_System_Array_GetLongLength (MonoArray *arr, gint32 dimension)
+ves_icall_System_Array_GetLongLength (MonoArrayHandle arr, gint32 dimension, MonoError *error)
 {
-	gint32 rank = m_class_get_rank (mono_object_class (&arr->obj));
-
-	if ((dimension < 0) || (dimension >= rank)) {
-		ERROR_DECL (error);
+	if (dimension < 0 || dimension >= m_class_get_rank (mono_handle_class (arr))) {
 		mono_error_set_index_out_of_range (error);
-		mono_error_set_pending_exception (error);
 		return 0;
 	}
-	
-	if (arr->bounds == NULL)
- 		return arr->max_length;
- 	
- 	return arr->bounds [dimension].length;
+
+	return MONO_HANDLE_GETVAL (arr, bounds) ? MONO_HANDLE_GETVAL (arr, bounds [dimension].length)
+						: MONO_HANDLE_GETVAL (arr, max_length);
 }
 
 ICALL_EXPORT gint32
-ves_icall_System_Array_GetLowerBound (MonoArray *arr, gint32 dimension)
+ves_icall_System_Array_GetLowerBound (MonoArrayHandle arr, gint32 dimension, MonoError *error)
 {
-	gint32 rank = m_class_get_rank (mono_object_class (&arr->obj));
+	icallarray_print ("%s arr:%p dimension:%d\n", __func__, arr, (int)dimension);
```

Comment:
So do you get output for each of the converted icalls when you run the new tests? And it all looks reasonable?